packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -489,7 +491,11 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
 
     final DeviceGestureSettings? gestureSettings = MediaQuery.maybeGestureSettingsOf(context);
 
-    final Set<WidgetState> states = <WidgetState>{if (!enabled) WidgetState.disabled};
+    final Set<WidgetState> states = <WidgetState>{
+      if (!enabled) WidgetState.disabled,
+      if (widget.onPressed != null || widget.onLongPress != null) WidgetState.pressed,
```

Comment:
Are you sure about this? If I understand correctly these are the callbacks, so you're evaluating whether the user wants to listen to the gestures, while the state is about whether the button is pressed.

Reply:
I wanted a clarification on that actually... so for the pressed state how should I determine it? Was confused on that...

Reply:
I think `_tapInProgress` is a perfect indicator. Although you might want to wrap every change of `_tapInProgress` in a `setState` because they currently aren't and you need setState to reevaluate the cursor.

Reply:
I see ```_tapInProgress``` was recently introduced... let me sync my local with upstream and use that to determine if the state is pressed or not. That would work right?

Reply:
No, just wrap every `_tapInProgress = xxx` in a setState.

Reply:
You still need to change this condition to `_tapInProgress`
```suggestion
      if(_tapInProgress) WidgetState.pressed,
```